examples/arduino/z_pull.ino

```diff
@@ -86,37 +74,51 @@ void setup() {
     }
     Serial.println("OK");
 
-    // Start the receive and the session lease loop for zenoh-pico
-    zp_start_read_task(z_session_loan_mut(&s), NULL);
-    zp_start_lease_task(z_session_loan_mut(&s), NULL);
-
-    // Declare Zenoh subscriber
-    Serial.print("Declaring Subscriber on ");
-    Serial.print(KEYEXPR);
-    Serial.println(" ...");
-    // @TODO
-    // z_owned_closure_sample_t callback;
-    // z_closure_sample(&callback, data_handler, NULL, NULL);
-    // @TODO
-    // z_view_keyexpr_t ke;
-    // z_view_keyexpr_from_str_unchecked(&ke, KEYEXPR);
-    // sub = z_declare_pull_subscriber(z_session_loan(&s), z_view_keyexpr_loan(&ke), z_closure_sample_move(&callback),
-    // NULL); if (!z_pull_subscriber_check(&sub)) {
-    //     Serial.println("Unable to declare subscriber.");
-    //     while (1) {
-    //         ;
-    //     }
-    // }
-    // Serial.println("OK");
-    // Serial.println("Zenoh setup finished!");
-    Serial.println("Pull Subscriber not supported... exiting");
+    if (zp_start_read_task(z_session_loan_mut(&s), NULL) < 0 || zp_start_lease_task(z_session_loan_mut(&s), NULL) < 0) {
+        printf("Unable to start read and lease tasks\n");
+        z_close(z_session_move(&s));
+        return;
+    }
+
+    printf("Declaring Subscriber on '%s'...\n", KEYEXPR);
+    z_owned_closure_sample_t closure;
+    z_ring_channel_sample_new(&closure, &handler, SIZE);
+    z_owned_subscriber_t sub;
+    z_view_keyexpr_t ke;
+    z_view_keyexpr_from_str(&ke, KEYEXPR);
+    if (z_declare_subscriber(&sub, z_session_loan(&s), z_view_keyexpr_loan(&ke), z_closure_sample_move(&closure),
+                             NULL) < 0) {
+        Serial.println("Unable to declare subscriber.");
+        return;
+    }
+
+    Serial.println("OK");
+    Serial.println("Zenoh setup finished!");
 
     delay(300);
 }
 
 void loop() {
-    delay(5000);
-    // z_subscriber_pull(z_pull_subscriber_loan(&sub));
+    z_owned_sample_t sample;
+    z_result_t res;
+    for (res = z_ring_handler_sample_try_recv(z_ring_handler_sample_loan(&handler), &sample); res == Z_OK;
+         res = z_ring_handler_sample_try_recv(z_ring_handler_sample_loan(&handler), &sample)) {
+        z_view_string_t keystr;
+        z_keyexpr_as_view_string(z_sample_keyexpr(z_sample_loan(&sample)), &keystr);
+        z_owned_string_t value;
+        z_bytes_deserialize_into_string(z_sample_payload(z_sample_loan(&sample)), &value);
+        Serial.print(">> [Subscriber] Pulled (");
+        Serial.print(z_string_data(z_view_string_loan(&keystr)));
```

Comment:
maybe it is better to use Serial.write(), in order to avoid convincing users that strings are always null-terminated ?